include/psa/crypto.h

```diff
@@ -57,6 +55,10 @@ extern "C" {
  * algorithms, key types, policies, etc. */
 #include "crypto_types.h"
 
+/* Include "crypto_platform.h" after "crypto_types.h" as it relies on
+ * `psa_key_id_t` defined by "crypto_types.h" */
+#include "crypto_platform.h"
```

Comment:
I am not sure what do you mean here. What in crypto_types.h relies on crypto_platform.h? My understanding is that crypto_types.h contains the types defined by the PSA spec thus I don't really get why something in this file would be platform dependent. This was the case when `psa_key_id_t` was redefined on a platform basis but this is one of the goal of this PR to avoid that.

Reply:
`psa_key_extended_id_t` (or whatever it ends up being called) has a definition that depends on the platform.

Reply:
This has been removed.